core/dbt/task/test.py

```diff
@@ -157,7 +157,7 @@ def execute(self, test: TestNode, manifest: Manifest):
             message = f"Got {num_errors}, configured to fail if {test.config.error_if}"
             failures = result.failures
         elif result.should_warn:
-            if get_flags().WARN_ERROR:
+            if get_flags().WARN_ERROR or get_flags().WARN_ERROR_OPTIONS.includes("LogTestResult"):
```

Comment:
Catching up on the discussion from the issue, I understand we're implementing this for a specific event, as opposed to more generally, because the `TestTask`  marks a test node as failed and only emits this particular `LogTestResult` event as an error if status is Fail. 👍 

Since this line and [TestRunner.print_result_line](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/task/test.py#L95-L110) will need to be updated together for this to work, I'd recommend adding a class attribute to TestRunner, e.g. `TestRunner._LOG_TEST_RESULT_EVENT = LogTestResult` and referencing it here as `get_flags().WARN_ERROR_OPTIONS.includes(self.LOG_TEST_RESULT_EVENT.__name__)`, and in as `fire_event(self.LOG_TEST_RESULT_EVENT, ... ).